### PR TITLE
support file base config

### LINKED ISSUE #5: remove hardcoding of kubernetes specific paths
Right now gazel is coded to the kubernetes repository. It should be possible to run gazel on other go repos without forking. A json or yaml config file in the repo root might work.

cc @justinsb since you may have already done this. @pjjw since I see you've forked gazel because this feature didn't exist.

### PR DESCRIPTION
**Pull request by mikedanese**
fixes #5

### CODE REVIEW

**Comment by rmmh** [comment]
(bump for gubernator)

File: config.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+)
+
+type Cfg struct {
+	GoPrefix string
+	// evalutated recursively, defaults to ["."]
```
Comment (ixdy) [inline config.go]: typo: evaluated

File: config.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	SrcDirs []string
+	// regexps that match packages to skip
+	SkippedPaths []string
+}
+
+func ReadCfg(root, cfgPath string) (*Cfg, error) {
+	if filepath.IsAbs(cfgPath) {
```
Comment (ixdy) [inline config.go]: is this logic inverted?

File: gazel.go [hunk 3]
```diff
@@ -48,7 +50,42 @@ func main() {
 }
 
 type Venderor struct {
```
Comment (ixdy) [inline gazel.go]: shouldn't this be `Vendorer` rather than `Venderor`, given that the path is `vendor/`?

I guess the ship has already sailed.

**Review by ixdy — COMMENTED** [review]
basically LG

File: gazel.go (hunk 3 above)
Comment (mikedanese) [reply gazel.go]: Fixed

File: config.go (hunk 2 above)
Comment (mikedanese) [reply config.go]: Thanks, Fixed

File: config.go (hunk 1 above)
Comment (mikedanese) [reply config.go]: Fixed

**Pull request merged**